lib/pq
# Adding sslcert and sslkey functionality

Allows specification of the cert and key used for an SSL Postgres connection. 
## To test

Update `postgresql.conf`
[how to find this file](http://stackoverflow.com/questions/3602450/where-are-my-postgres-conf-files)

```
...
ssl = on                # (change requires restart)
ssl_cert_file = '{PQ_DIR}/certs/posgresql.crt'
ssl_key_file = '{PQ_DIR}/certs/posgresql.key'
ssl_ca_file = '{PQ_DIR}/certs/root.crt'
```

Update `pg_hba.conf`

```
hostssl    pqgotest    postgres  127.0.0.1/32         cert
```

> make sure to comment out all other lines
### Restart postgres

``` shell
pg_ctl restart -D /usr/local/var/postgres
```
### Setup .postgresql

Link `~/.postgresql` to `certs` dir

``` shell
ln -s {PQ_DIR}/certs ~/.postgresql 
```
### Run tests

``` bash
PGSSLMODE=require PGUSER=postgres go test 
```
